src/sim/eventq.cc

```diff
@@ -223,7 +209,6 @@ EventQueue::remove(Event *event)
 Event *
 EventQueue::serviceOne()
 {
-    std::lock_guard<EventQueue> lock(*this);
```

Comment:
This is definitely incorrect. Removing this line removes the mutex that protects the event queue in multi-threaded simulation.

Reply:
Agreed, will revert